web/packages/design/src/Dialog/Dialog.tsx

```diff
@@ -16,42 +16,45 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from 'styled-components';
+import { forwardRef, PropsWithChildren } from 'react';
+import styled, { StyleFunction } from 'styled-components';
 
-import Modal from './../Modal';
+import Modal, { Props as ModalProps } from './../Modal';
 
-export default class Dialog extends React.Component {
-  render() {
-    const { children, dialogCss, className, ...modalProps } = this.props;
-    return (
-      <Modal role="dialog" {...modalProps}>
-        <ModalBox>
-          <DialogBox
-            data-testid="dialogbox"
-            dialogCss={dialogCss}
-            className={className}
-          >
-            {children}
-          </DialogBox>
-        </ModalBox>
-      </Modal>
-    );
-  }
-}
+// ModalProps enforces its own requirements with regards to the children prop through types
+// which we do not care about here, as children passed to <Dialog> are not passed directly to <Modal>.
+type ModalPropsWithoutChildren = Omit<ModalProps, 'children'>;
 
-Dialog.defaultProps = {
-  disableBackdropClick: true,
-  disableEscapeKeyDown: true,
-};
-
-Dialog.propTypes = {
-  ...Modal.propTypes,
-  children: PropTypes.node,
-  dialogCss: PropTypes.func,
-  className: PropTypes.string,
-};
+export const Dialog = forwardRef<
+  HTMLDivElement,
+  PropsWithChildren<
+    {
+      className?: string;
+      dialogCss?: StyleFunction<any>;
+    } & ModalPropsWithoutChildren
+  >
+>((props, ref) => {
+  const { children, dialogCss, className, ...modalProps } = props;
+  return (
+    <Modal disableBackdropClick disableEscapeKeyDown {...modalProps}>
+      <ModalBox>
+        {/*
+            ref is supposed to be set on DialogBox, not Modal, because when used with
+            react-transition-group, it's DialogBox that's going to have its className set based on
+            the transition state.
+          */}
```

Comment:
Passing a ref to `<Dialog>` was needed because in teleport.e, [`ReviewRequests` has a transition on `NotificationRoutingRulesDialog`](https://github.com/gravitational/teleport.e/pull/5023/files#diff-5e9a8e166ee07cdf201ef80f7268d94e3699a81410947bb443959de133abcce6) which then [renders a `Dialog`](https://github.com/gravitational/teleport.e/pull/5023/files#diff-a664fc40de9fd59cfa28fc83f4724d951b18cc4c781fad935878f501f4de2bed).